documents/model.go

```diff
@@ -433,9 +441,10 @@ func (m *CoreDocumentModel) CalculateSigningRoot(dataRoot []byte) error {
 // AccountCanRead validate if the core document can be read by the account .
 // Returns an error if not.
 func (m *CoreDocumentModel) AccountCanRead(account identity.CentID) bool {
-	// loop though read rules
-	return m.findRole(coredocumentpb.Action_ACTION_READ_SIGN, func(role *coredocumentpb.Role) bool {
-		return isAccountInRole(role, account)
+	// loop though read rules, check all the rules
+	return m.findRole(nil, func(_, _ int, role *coredocumentpb.Role) bool {
+		_, found := isAccountInRole(role, account)
```

Comment:
Wouldn't this change lead to any extra read rule to also match? Shouldn't we explicitly check that it's read or read_sign?

Reply:
yeap. Agreed and changed